pkg/controller/operators/olm/operator.go

```diff
@@ -1782,6 +1782,11 @@ func (a *Operator) updateInstallStatus(csv *v1alpha1.ClusterServiceVersion, inst
 		return nil
 	}
 
+	if err := findFirstError(k8serrors.IsServiceUnavailable, strategyErr, apiServiceErr, webhookErr); err != nil {
+		csv.SetPhaseWithEventIfChanged(v1alpha1.CSVPhasePending, v1alpha1.CSVReasonDetectedClusterChange, err.Error(), now, a.recorder)
```

Comment:
I'm not sure if this Phase and Reason makes sense. If a CSV is in phase Successful when this error occurs (which I think is generally the case during cluster upgrade scenarios), it looks like resetting to Pending will cause OLM to step through InstallReady and Installing before getting back to Succeeded.

Is there a better Phase and/or Reason to use? I couldn't find any other areas of OLM that deal with transient issues like this. Is it possible we need a new Phase or Reason?

Reply:
I don't think we want transient errors to result in any updates to the CSV. Instead, the sync function can return a non-nil error and should be automatically retried later.

Reply:
That seems reasonable. The only concern I have with this approach is handling a "transient" error that's actually permanent for some reason.

In that case, there would never be an indication on the CSV that anything is wrong.

In practice a kube-apiserver that's unavailable would probably mean no user could actually query the CSV to know there was an issue AND they would probably be more worried about the kube-apiserver than the CSV.

Reply:
@benluddy Updated to return the error without changing CSV phase/reason.